leaf-ui/rappid-extensions/BackboneModelsIntention.js

```diff
@@ -39,29 +39,468 @@ var EvolvingFunctionBBM = Backbone.Model.extend({
+     * the global model variable, if such absolute Constraint object(s) exists
+     */
+
+    removeAbsConstraint: function() {
+        var i = 0;
+        while (i < graph.constraints.length) {
+            var constraint = graph.constraints.models[i];
+            if (constraint.get('type') == 'A' && constraint.get('srcID') === this.get('id')) {
+                graph.constraints.splice(i, 1);
+            } else {
+                i++;
+            }
+        }
+    }, 
+ 
+    /**
+     * Adds a new FuncSeg to the end of this Intention's EvolvingFunction's
+     * function list.
+ 
+     * Also adds a corresponding Constraint object reprenting an absolute constraint
+     * into the global model variable.
+     *
+     *This function should only be used to add new function
+     * segments for user defined functions
+     *
+     * @param {String} funcType
+     *   ex: 'C'
+     * @param {String} satValue
+     *   ex: '0000'
+     */
+    addUserDefinedSeg: function(funcType, satValue){
+ 
+        var len = this.evolvingFunction.getFuncSegments().length;
+        // originally used to be funcStop 
+        var start = this.evolvingFunction.getFuncSegments()[len - 1].get('stopTP');
+        var code = start.charCodeAt(0) + 1;
+        var stop = String.fromCharCode(code);
+ 
+        //create new funcsegment model 
+        var new_model = new FunctionSegmentBBM({type: funcType, refEvidencePair: satValue, startTP: start, stopTP: stop}); 
+        this.evolvingFunction.getFuncSegments().push(new_model); 
+ 
+        model.constraints.push(new ConstraintBBM({type: 'A', srcID: this.get('id'), srcRefTP: start, destID: null, destRefTP: null}));
+ 
+    }, 
+    /**
+     * Sets the marked value for the FuncSegments in the
+     * EvolvingFunction for this Intention
+     *
+     * This function will only be called for I, D, RC, MP, MN functions
+     */
+    setMarkedValueToFunction: function(satValue) {
+        var funcType = this.evolvingFunction.get('type');
+ 
+        var len = this.getFuncSegments().length;
+        this.getFuncSegments()[len - 1].set('refevidencePair', satValue);
+ 
+        if (funcType == 'MP' || funcType == 'MN') {
+            this.getFuncSegments()[0].set('refEvidencePair', satValue);
+        }
+    }, 
+ 
+    /**
+     * Sets the function type and marked value for the
+     * last FuncSegment for this Intention's EvolvingFunction
+     *
+     * @param {String} funcValue
+     */
+    setUserDefinedSegment: function(funcValue) {
+        var funcSegLen = this.getFuncSegments().length;
+        var functionSegment = this.getFuncSegments()[funcSegLen - 1];
+        functionSegment.set('type', funcValue); 
+        if (funcValue == 'C' || funcValue =='R') {
+            funcSeg.set('refEvidencePair', '0000');
+        } 
+    }, 
+ 
+    /**
+     * Sets the satisfaction value for the last function segment
+     * in this Intention's evolving function, to satVal
```

Comment:
Will this be part of the Function Segment BBM?

Reply:
This function was never called so we just deleted it..